packages/nodes-base/nodes/Google/CloudStorage/BucketDescription.ts

```diff
@@ -0,0 +1,617 @@
+						},
+					},
+					send: {
+						paginate: true,
+						preSend: [
+							async function (this, requestOptions) {
+								if (!requestOptions.qs) requestOptions.qs = {};
+								const returnAll = this.getNodeParameter('returnAll') as boolean;
+
+								if (!returnAll) {
+									const key = this.getNodeParameter('projection') as string;
+									requestOptions.qs.maxResults =
+										key === 'noAcl' ? PAGE_LIMITS.noAcl : PAGE_LIMITS.full;
+								}
+								return requestOptions;
+							},
+						],
+					},
+					operations: {
+						async pagination(this, requestOptions) {
+							if (!requestOptions.options.qs) requestOptions.options.qs = {};
+							let executions: INodeExecutionData[] = [];
+							let responseData: INodeExecutionData[];
+							let nextPageToken: string | undefined = undefined;
+							const returnAll = this.getNodeParameter('returnAll') as boolean;
+
+							do {
+								requestOptions.options.qs.pageToken = nextPageToken;
+								responseData = await this.makeRoutingRequest(requestOptions);
+
+								// Check for another page
+								const lastItem = responseData[responseData.length - 1].json;
+								nextPageToken = lastItem.nextPageToken as string | undefined;
+
+								// Extract just the list of buckets from the page data
+								responseData.forEach((page) => {
+									const buckets = page.json.items as IDataObject[];
+									if (buckets) {
+										executions = executions.concat(buckets.map((bucket) => ({ json: bucket })));
+									}
+								});
+								// If we don't return all, just return the first page
+							} while (returnAll && nextPageToken);
+
+							// Return all execution responses as an array
+							return executions;
+						},
+					},
+				},
+				action: 'Get all Buckets',
+			},
+			{
+				name: 'Update',
+				value: 'update',
+				description: 'Update the metadata of a bucket',
+				routing: {
+					request: {
+						method: 'PATCH',
+						url: '={{"/b/" + $parameter["bucketName"]}}',
+						qs: {
+							project: '={{$parameter["projectId"]}}',
+							projection: '={{$parameter["projection"]}}',
+						},
+						body: {},
+						returnFullResponse: true,
+					},
+					send: {
+						preSend: [
+							async function (this, requestOptions) {
+								if (!requestOptions.qs) requestOptions.qs = {};
+								const additionalQs = this.getNodeParameter('createAcl') as IDataObject;
+								const filters = this.getNodeParameter('getFilters') as IDataObject;
+
+								// Merge in the options into the queryset
+								requestOptions.qs = Object.assign(requestOptions.qs, additionalQs, filters);
+								return requestOptions;
+							},
+						],
+					},
+				},
+				action: 'Create a new Bucket',
+			},
+		],
+		default: 'getAll',
+	},
+];
+
+export const bucketFields: INodeProperties[] = [
+	{
+		displayName: 'Project ID',
+		name: 'projectId',
+		type: 'string',
+		required: true,
+		placeholder: 'Project ID',
+		displayOptions: {
```

Comment:
I've noticed this argument is sent on every request as query string with the same name, so we can simplify this:
```suggestion
		displayName: 'Project ID',
		name: 'projectId',
		type: 'string',
		required: true,
		placeholder: 'Project ID',
				routing: {
			request: {
				qs: {
					project: '={{ $value }}',
				},
			},
		},
		displayOptions: {
```